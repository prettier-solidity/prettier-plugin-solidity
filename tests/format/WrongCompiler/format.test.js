// Should warn twice
runFormatTest(import.meta, ['solidity-parse'], { compiler: '0.8.4' });
// Should warn once
runFormatTest(import.meta, ['solidity-parse'], {
  compiler: 'v0.7.5-nightly.2020.11.9+commit.41f50365'
});
// Should not warn
runFormatTest(import.meta, ['solidity-parse'], {
  compiler: 'v0.7.3+commit.9bfce1f6'
});
