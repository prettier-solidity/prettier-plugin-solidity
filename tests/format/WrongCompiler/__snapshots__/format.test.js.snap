// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrongCompiler.sol - {"compiler":"0.8.4"} format 1`] = `
====================================options=====================================
compiler: "0.8.4"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
pragma solidity >= 0.7.0;
pragma solidity >0.7.2 <= 0.7.5;
pragma solidity 0.7.3;
=====================================output=====================================
pragma solidity >=0.7.0;
pragma solidity >0.7.2 <=0.7.5;
pragma solidity 0.7.3;

================================================================================
`;

exports[`WrongCompiler.sol - {"compiler":"v0.7.3+commit.9bfce1f6"} format 1`] = `
====================================options=====================================
compiler: "v0.7.3+commit.9bfce1f6"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
pragma solidity >= 0.7.0;
pragma solidity >0.7.2 <= 0.7.5;
pragma solidity 0.7.3;
=====================================output=====================================
pragma solidity >=0.7.0;
pragma solidity >0.7.2 <=0.7.5;
pragma solidity 0.7.3;

================================================================================
`;

exports[`WrongCompiler.sol - {"compiler":"v0.7.5-nightly.2020.11.9+commit.41f50365"} format 1`] = `
====================================options=====================================
compiler: "v0.7.5-nightly.2020.11.9+commit.41f50365"
parsers: ["solidity-parse"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
pragma solidity >= 0.7.0;
pragma solidity >0.7.2 <= 0.7.5;
pragma solidity 0.7.3;
=====================================output=====================================
pragma solidity >=0.7.0;
pragma solidity >0.7.2 <=0.7.5;
pragma solidity 0.7.3;

================================================================================
`;
